geometric_kernels/frontends/pytorch/gpytorch.py

```diff
@@ -1,63 +1,96 @@
 """
-Pytorch kernel wrapper
+GPyTorch kernel wrapper
 """
 import gpytorch
-import numpy as np
 import torch
 
 from geometric_kernels.kernels import BaseGeometricKernel
 from geometric_kernels.spaces.base import Space
 
 
-class GPytorchGeometricKernel(gpytorch.kernels.Kernel):
+class GPyTorchGeometricKernel(gpytorch.kernels.Kernel):
     """
-    Pytorch wrapper for `BaseGeometricKernel`
+    GPyTorch wrapper for `BaseGeometricKernel`
+
+    Note: you cannot change trainable_nu after constructing the object.
     """
 
     has_lengthscale = True
 
     def __init__(
         self,
-        kernel: BaseGeometricKernel,
-        lengthscale=1.0,
-        nu=np.inf,
+        base_kernel: BaseGeometricKernel,
+        lengthscale=None,
+        nu=None,
+        variance=1.0,
         trainable_nu: bool = False,
         **kwargs,
     ):
         super().__init__(**kwargs)
 
-        self._kernel = kernel
+        self.base_kernel = base_kernel
+
+        default_params = base_kernel.init_params()
+
+        if nu is None:
+            nu = default_params["nu"]
 
-        self.lengthscale = torch.tensor(lengthscale)
+        if lengthscale is None:
+            lengthscale = default_params["lengthscale"]
 
-        self.trainable_nu = trainable_nu
-        if self.trainable_nu and torch.isinf(nu):
+        lengthscale = torch.as_tensor(lengthscale)
+        variance = torch.as_tensor(variance)
+        nu = torch.as_tensor(nu)
+
+        self._trainable_nu = trainable_nu
+        if self._trainable_nu and torch.isinf(nu):
             raise ValueError("Cannot have trainable `nu` parameter with infinite value")
 
-        if self.tranable_nu:
+        self.lengthscale = lengthscale
+
+        self.register_parameter(
```

Comment:
It's been a long time since I used gpytorch, is it not a general practice to have unit-variance kernels and provide scaling by wrapping them in a `gpytorch.kernels.ScaleKernel`? I vaguely remember this as a reason why the Gpytorch frontend kernel didn't have a variance parameter in the first place.

Reply:
Done